### PR TITLE
333 tags component

### LINKED ISSUE #333: [FRONTEND] Create tags component
**Is your feature request related to a problem? Please describe.**

Note a user should be able to pass in an `onClose` handler

Image

**BEFORE MERGING**

- [x] Storybooks created where possible
- [ ] Tested with [nock](https://tanstack.com/query/v3/docs/framework/react/guides/testing) if data fetching
- [ ] Tests written for critical interactions
- [ ] PR Reviewed (For non-trivial changes)
- [ ] Changes tested after rebasing on master or merging in master (_hint_: `git fetch origin master:master`, then `git rebase master` or `git merge master`)
- [ ] All required PR checks passing

### PR DESCRIPTION
**Pull request by zlrkw11**
closes #333

### CODE REVIEW

**Comment by zlrkw11** [comment]
image
- normal tag (default displaying tag)

image
- interactive tag (closing functionality implemented)

File: client/src/components/generic/TagsComponent/TagsComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    <div className="bg-light-blue-60 flex h-[32px] w-[80px] items-center justify-center overflow-hidden whitespace-nowrap rounded-full">
+      <h5 className="font-bold">{children}</h5>
+    </div>
+  )
+}
+
+const InteractiveTagsComponent = ({ children }: props) => {
+  const [isVisible, setIsVisible] = useState(true)
+  const handleClose = () => {
+    setIsVisible(false)
```
Comment (bcho892) [inline client/src/components/generic/TagsComponent/TagsComponent.tsx:24]: gj on this one, another note tho can we let consumers pass in a custom handler like `onClose` type `() => void` (i.e we might be using this to display filters, and when we close one of the tags we want to remove the filter)

File: client/src/components/generic/TagsComponent/TagsComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  }
+  if (!isVisible) {
+    return null
+  }
+
+  return (
+    <div className="bg-dark-blue-100 flex h-[32px] w-[80px] items-center justify-center overflow-hidden whitespace-nowrap rounded-full">
```
Comment (bcho892) [inline client/src/components/generic/TagsComponent/TagsComponent.tsx:31]: We should maybe have an `onClick` handler for this

**Review by bcho892 — CHANGES_REQUESTED** [review]
Overall pretty clean, but for both tags we should allow the consumer to provide a handler for `onClick` and `onClose` so we can have custom functionality.

**Review by bcho892 — APPROVED** [review]

**Pull request merged**